### PR TITLE
Kotlin Flex Sync Template App updates related to the tutorial

### PR DESCRIPTION
**Pull request by mongomoe**
This PR implements the following:
* Changes 'MongoDB Realm Sync' -> 'Atlas Device Sync'
* Changes 'Todo' -> 'Item'
* Recreate the original [Dialog](https://developer.android.com/guide/topics/ui/dialogs) (i.e Pop-Up/Modal) to be more dynamic. I moved the dialog to its own XML layout file, so that the UI could be generated with XML rather than with Kotlin code. This gives the tutorial reader a simpler experience to adding a Priority Spinner (i.e, dropdown) to the Dialog. You can view a PR where I've added a Priority spinner to this code (to mimic the tutorial readers experience) here: https://github.com/mongomoe/realm-template-apps/pull/2 
* Adds an [Android Toast](https://developer.android.com/guide/topics/ui/notifiers/toasts) as simple feedback alerting the template app user that the item has been created and the item's summary. The Android Toast feedback popup appears for a few seconds before disappearing.

Also, here are some screenshots:

### Dialog (as you can see visually it hasn't changed from Nate's version):
Screen Shot 2022-08-17 at 7 22 11 PM

### The Android Toast Feedback displaying that an item was created: 
Screen Shot 2022-08-17 at 7 22 43 PM

### The Item screen after the Android Toast Feedback fades away (as you can see it has not changed at all from Nate's version):
Screen Shot 2022-08-17 at 7 23 08 PM

### CODE REVIEW

File: kotlin-sdk/app/src/main/res/layout/activity_todo.xml [hunk 1]
```diff
@@ -4,10 +4,10 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
```
Comment (nlarew) [inline kotlin-sdk/app/src/main/res/layout/activity_todo.xml:5]: Should we update this file name to `activity_item.xml` + rename when we reference it?

**Review by nlarew — APPROVED** [review]
LGTM w/ a suggestion

File: kotlin-sdk/app/src/main/java/com/mongodb/app/ItemActivity.kt [hunk 2]
```diff
@@ -73,22 +72,22 @@ class TodoActivity : AppCompatActivity() {
...
+            recyclerView.adapter = itemAdapter
         }
     }
 
     /**
-     * Add buttons to the todo menu.
+     * Add buttons to the item menu.
      */
     override fun onCreateOptionsMenu(menu: Menu): Boolean {
         menuInflater.inflate(R.menu.todo_menu, menu)
```
Comment (nlarew) [inline kotlin-sdk/app/src/main/java/com/mongodb/app/ItemActivity.kt:85]: Do we need to update this to `R.menu.item_menu`?

File: kotlin-sdk/app/src/main/java/com/mongodb/app/ItemAdapter.kt [hunk 3]
```diff
@@ -56,58 +50,58 @@ internal class TodoAdapter(private var data: RealmResults<Todo>, private val rea
...
         var name: TextView = view.findViewById(R.id.name)
         var menu: TextView = view.findViewById(R.id.popup_menu)
         var checkbox: CheckBox = view.findViewById(R.id.checkbox)
     }
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TaskViewHolder {
-        val todoView = LayoutInflater.from(parent.context).inflate(R.layout.todo_view, parent, false)
-        return TaskViewHolder(todoView)
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
+        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.todo_view, parent, false)
```
Comment (nlarew) [inline kotlin-sdk/app/src/main/java/com/mongodb/app/ItemAdapter.kt:61]: Do we need to update this to `R.layout.item_layout`?

File: kotlin-sdk/app/src/main/res/layout/activity_todo.xml (hunk 1 above)
Comment (mongomoe) [reply kotlin-sdk/app/src/main/res/layout/activity_todo.xml:5]: Yes, good catch - prior to this I replaced 'todo' with 'item' for the in-code usages, but hadn't renamed the files + file references. I've done that now.

File: kotlin-sdk/app/src/main/java/com/mongodb/app/ItemActivity.kt [hunk 4]
```diff
@@ -115,31 +115,40 @@ class TodoActivity : AppCompatActivity() {
...
-                        dialog.dismiss()
-                        val todo = Todo(realmApp.currentUser!!.identity)
-                        todo.summary = input.text.toString()
-                        CoroutineScope(Dispatchers.IO).launch {
-                            realm.write {
-                                this.copyToRealm(todo)
-                            }
+        val mBuilder = AlertDialog.Builder(this)
+            .setMessage("Enter Item Name:")
+        val mView: View = layoutInflater.inflate(R.layout.create_item_dialog, null)
```
Comment (nlarew) [inline kotlin-sdk/app/src/main/java/com/mongodb/app/ItemActivity.kt:123]: Just curious, what's the `m` stand for here?

File: kotlin-sdk/app/src/main/java/com/mongodb/app/ItemActivity.kt [hunk 5]
```diff
@@ -115,31 +115,40 @@ class TodoActivity : AppCompatActivity() {
...
+                        Toast.LENGTH_SHORT
+                    ).show()
                 }
             .setNegativeButton("Cancel") { dialog, _ ->
                     dialog.cancel()
                 }
         val dialog = dialogBuilder.create()
         dialog.setView(input)
-        dialog.setTitle("Add ToDo")
+        dialog.setTitle("Add Item")
```
Comment (nlarew) [inline kotlin-sdk/app/src/main/java/com/mongodb/app/ItemActivity.kt:151]: FWIW in the RN app I have this dialog titled as "Add To-Do Item" + left the button as "Add To-Do" + titled the screen "Your To-Do Items"

**Review by nlarew — APPROVED** [review]
LGTM with a couple comments

**Pull request merged**